akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/ClusterStatusCheck.scala

```diff
@@ -23,15 +23,30 @@ import scala.concurrent.{ ExecutionContext, Future }
 
 class ClusterStatusCheck extends ReadinessCheck with HealthCheck {
   def healthy(actorSystem: ExtendedActorSystem)(implicit ec: ExecutionContext): Future[Boolean] =
-    isUp(actorSystem)
+    status(actorSystem) map {
+      case MemberStatus.Joining  => true
+      case MemberStatus.WeaklyUp => true
+      case MemberStatus.Up       => true
+      case MemberStatus.Leaving  => true
+      case MemberStatus.Exiting  => true
+      case MemberStatus.Down     => false
+      case MemberStatus.Removed  => false
+    }
 
   def ready(actorSystem: ExtendedActorSystem)(implicit ec: ExecutionContext): Future[Boolean] =
-    isUp(actorSystem)
+    status(actorSystem) map {
+      case MemberStatus.Joining  => false
```

Comment:
How is this going to be used?
One can configure [min-nr-of-members](https://doc.akka.io/docs/akka/current/cluster-usage.html?language=scala#how-to-startup-when-cluster-size-reached) meaning that joining members will not be moved to Up until this number of members exists (has joined) in the Cluster.

Why is a Joining member not ready?

Reply:
> Why is a Joining member not ready?

Readyness is mostly for "send prod traffic to this node", which I think we want to delay until it has joined and is Up. We want to make it talk to other nodes in the cluster until then though, which makes the bootstrap trickier, in the sense that we need to use k8s APIs as until then they don't show up in DNS AFAIR.

Reply:
The section about it is here (it's short) https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#define-readiness-probes

Reply:
What K said + you can set up a service in k8s so that you do or do not see members in DNS until they are ready.

Looking into it some more I  think we should not put akka management / remoting ports on the same k8s service as the prod traffic via gRPC/HTTP etc as you typically want the opposite behavior

Reply:
Sorry for my ignorance. If the first seed joins itself without being moved to Up how can other nodes discover it and join if it doesn't show up in DNS? Perhaps this is the whole thing that Christopher is saying with the separation of HTTP and remoting.

If you go forward with this and it doesn't work with `min-nr-of-members > 1` it must come with a fat warning in the documentation.

Reply:
The current recommended bootstrap implementation for Kubernetes uses the Kubernetes API to detect the nodes, so doesn't depend on DNS.

@chbatey said that you can configure the option for DNS to include/exclude pods with failing readiness probes on a per-service basis. I can't find a reference for this in the Kubernetes documentation, though... can you add a link?

Reply:
Also, this pull request is _broadening_ the definition of "ready" to include `WeaklyUp`, and "healthy" (or "live" in k8s terminology) to include everything that isn't `Down` or `Removed`, not limiting them further. Assuming that this has been working with `min-nr-of-members > 1` all along, it does not seem to me like this would regress that.

This seems like a good change. The problem it's solving is that Kubernetes will kill containers that are not healthy/live for a certain period of time, and this was causing clusters that were taking a long time to bootstrap to be killed, if the initial delay for the liveness probe wasn't adjusted to exceed the `stable-margin`.